packages/gatsby/src/services/initialize.ts

```diff
@@ -52,6 +53,8 @@ export async function initialize({
     reporter.panic(`Missing program args`)
   }
 
+  detectHotLoaderToUse()
```

Comment:
I would like these to be super long and meaningful cause it's not just detecting because I would assume a `const hotLoader = detectHotLoaderToUse()` instead
```suggestion
  detectAndSetWhichReactHotLoaderToUse()
```